packages/iris-grid/src/IrisGridUtils.ts

```diff
@@ -790,11 +790,11 @@ class IrisGridUtils {
     return (
       sorts
         .map(sort => {
-          const { column: columnIndex, isAbs, direction } = sort;
+          const { columnName, isAbs, direction } = sort;
           if (direction === TableUtils.sortDirection.reverse) {
             return dh.Table.reverse();
           }
-          const column = IrisGridUtils.getColumn(columns, columnIndex);
+          const column = IrisGridUtils.getColumnByName(columns, columnName);
```

Comment:
This is correct, but we're going to need a migration strategy for data that was stored in the old format (testing against Enterprise). Can just check against the sort value to see if `columnName` is provided, or instead define the `DehydratedSort` type as `column: ModelIndex | ColumnName`, then have a typeof check and look up the column name if a `ModelIndex` is provided.

An example is how we handle `movedColumns`: https://github.com/deephaven/web-client-ui/blob/d6c8a988d62157abfb8daadbff5db3eaef21a247/packages/iris-grid/src/IrisGridUtils.ts#L236
@mattrunyon implemented that migration behaviour in https://github.com/deephaven/web-client-ui/pull/642/files (though it should be typed to a `ColumnName` instead of a `string`).

Reply:
The `movedColumns` wasn't really a migration I'd say. It was adding behavior (support for moving ranges in 1 move), not changing or removing any. This issue removes using `ModelIndex` for the sort column.

We'll still need to make sure the `ModelIndex` order is saved since we rely on that and not names for many things still. I'm not sure if that is accurate, but this ticket does remove the direct uses of `column.index` from the API. We still have some reliance on the column order being maintained by the engine which is a bigger issue to deal with.

Related #833 and #643

Reply:
I've updated the code to support index or column name when importing a sort, but exporting uses column name.